### PR TITLE
Add non-interval restart capability

### PR DESCRIPTION
**Pull request by DeniseWorthen**
Adds a configuration variable ``restart_fh`` to allow restarts on non-interval hours. See Discussion in UWM Issue [#1976](https://github.com/ufs-community/ufs-weather-model/issues/1976) for details.

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
@jiandewang We might want to wrap this PR into UWM https://github.com/ufs-community/ufs-weather-model/pull/2085. Is that OK?

**Comment by DeniseWorthen** [comment]
@jiandewang Can you review? That way we'll be ready at the UWM level. Thanks.

**Review by jiandewang — APPROVED** [review]
look good and test runs are fine.

**Comment by jiandewang** [comment]
@junwang-noaa can you review this PR ?

File: config_src/drivers/nuopc_cap/mom_cap.F90 [hunk 1]
```diff
@@ -1791,25 +1809,34 @@ end subroutine ModelAdvance
 
 
 subroutine ModelSetRunClock(gcomp, rc)
+
+  use ESMF, only : ESMF_TimeIntervalSet
```
Comment (junwang-noaa) [inline config_src/drivers/nuopc_cap/mom_cap.F90:1813]: Do we need this line?

Comment (DeniseWorthen) [reply config_src/drivers/nuopc_cap/mom_cap.F90:1813]: Yes, it is not used anywhere else and we specify only used ESMF procedures.

**Comment by zach1221** [comment]
Testing is complete on UFS-WM PR #2085 . Please feel free to merge this PR. @jiandewang @DeniseWorthen

**Comment by jiandewang** [comment]
@zach1221 MOM6 merged

**Pull request merged**